text/0000-argument-lifetimes.md

```diff
@@ -0,0 +1,492 @@
+- Feature Name: argument_lifetimes
+- Start Date: 2017-08-17
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Improves the clarity, ergonomics, and learnability around explicit lifetimes, so
+that instead of writing
+
+```rust
+fn two_args<'a, 'b>(arg1: &'a Foo, arg2: &'b Bar) -> &'b Baz
```

Comment:
Or if you prefer, you could find some other way to incorporate the `'a` so that it *does* have to be named.

For example:
 * Before: `fn two_args<'a, 'b: 'a>(arg1: &'a Foo, arg2: &'b Bar) -> &'a Baz<'b>`
 * After: `fn two_args(arg1: &'a Foo, arg2: &'b Bar) -> &'a Baz<'b>`